test/runtests.jl

```diff
@@ -116,6 +116,13 @@ push!(tests, "NMF" => ()->begin
         @test_skip "NMF"
     end
 end)
+push!(tests, "MIOpen" => ()->begin
+    if AMDGPU.functional(:MIOpen)
+        include("dnn/miopen.jl")
```

Comment:
I like the move from the `rocarray` directory to a dedicated `dnn` directory! For future PRs to the other libraries (rocBLAS et. al), feel free to also move them to their own folders, for consistency.